proto/kvrpcpb.proto

```diff
@@ -241,6 +255,31 @@ message BatchRollbackResponse {
     KeyError error = 2;
 }
 
+// CheckTxnStatusRequest checks the status of a transaction.
+// If the TTL of the transaction is exhausted, abort that transaction.
+// Otherwise, returns the TTL information in addition to commit/rollback.
+// CheckTxnStatusRequest also pushes the minCommitTS forword for large transaction.
+//
+// In the future, TiDB may not store TTL in the secondary lock, and it calls this API to
+// get the TTL from the primary key.
+//
+message CheckTxnStatusRequest {
+    Context context = 1;
+    bytes primary_key = 2;
+    uint64 start_version = 3;
```

Comment:
Is it`primary_key`'s `start_version`? Does it need a `read_version` to update lock's `min_commit_ts`?

Reply:
No, it's current transaction's start_version.

Reply:
How can we get commit info if we don't know the txn's `start_ts`?

Reply:
You are right, we need another ts.

Reply:
Yes, we need a ts to push forward the minCommitTS, and I add a new field here:
https://github.com/pingcap/kvproto/pull/443/files#diff-93fb3a82615ef50b8b76520198e3d36fR270

This may be the (startTS + 1) of the checker, or the cached ts in TiDB, or the real ts from PD.